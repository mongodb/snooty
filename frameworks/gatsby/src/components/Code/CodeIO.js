import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { cx, css as LeafyCss } from '@leafygreen-ui/emotion';
import Icon from '@leafygreen-ui/icon';
import Button from '@leafygreen-ui/button';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { palette } from '@leafygreen-ui/palette';
import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
import Input from '../Code/Input';
import Output from '../Code/Output';
import { isOfflineDocsBuild } from '../../utils/is-offline-docs-build';
import { OFFLINE_BUTTON_CLASSNAME, OFFLINE_OUTPUT_CLASSNAME } from '../../utils/head-scripts/offline-ui/code';
import { baseCodeStyle, borderCodeStyle } from './styles/codeStyle';

const outputButtonStyling = LeafyCss`
  padding: 0px;
  height: 24px;
  margin: 8px;

  ${
    isOfflineDocsBuild &&
    `
    span:nth-of-type(2) {
      display: none;
    }
    &.offline-hide-output {
      svg {
        transform: rotate(180deg);
      }
      span:nth-of-type(2) {
        display: inherit;
      }
      span:nth-of-type(1) {
        display: none;
      }
    }  
  `
  }
`;

const getButtonText = (showOutput) => (showOutput ? 'HIDE OUTPUT' : 'VIEW OUTPUT');

const outputContainerStyle = (showOutput) => LeafyCss`
  ${!showOutput && 'display: none;'}
`;

const CodeIO = ({ nodeData: { children }, ...rest }) => {
  const { darkMode } = useDarkMode();
  const needsIOToggle = children.length === 2;
  const onlyInputSpecified = children.length === 1;

  let initialOutputVisibility = true;
  if (needsIOToggle && children[1]?.options?.visible !== undefined) {
    initialOutputVisibility = !!children[1].options.visible;
  }
  const [showOutput, setShowOutput] = useState(() => (isOfflineDocsBuild ? true : initialOutputVisibility));
  const buttonText = getButtonText(showOutput);
  const arrow = showOutput ? 'ChevronUp' : 'ChevronDown';
  const outputBorderRadius = !showOutput ? '12px' : '0px';
  const singleInputBorderRadius = onlyInputSpecified ? '12px' : '0px';

  const handleClick = () => {
    setShowOutput((val) => !val);
  };

  if (children.length === 0) {
    return null;
  }

  return (
    <div
      css={css`
        ${baseCodeStyle}
        // Inner div of LG component has a width set to 700px. Unset this as part of our
        // override for docs when the language switcher is being used.
        > div > div {
          border-bottom-right-radius: ${singleInputBorderRadius};
          border-bottom-left-radius: ${singleInputBorderRadius};
        }

        // Controls output code block and toggle view bar style
        > div {
          border-bottom-right-radius: ${outputBorderRadius};
          border-bottom-left-radius: ${outputBorderRadius};
          margin: 0px;
        }
      `}
    >
      {needsIOToggle && (
        <>
          <Input nodeData={children[0]} />
          <IOToggle style={{ '--border-color': darkMode ? palette.gray.dark2 : palette.gray.light2 }}>
            <Button
              role="button"
              className={cx(outputButtonStyling, isOfflineDocsBuild ? OFFLINE_BUTTON_CLASSNAME : '')}
              disabled={false}
              onClick={handleClick}
              leftGlyph={<Icon glyph={arrow} fill="#FF0000" />}
            >
              <span>{buttonText}</span>
              {isOfflineDocsBuild && <span>{getButtonText(false)}</span>}
            </Button>
          </IOToggle>
          <div className={cx(outputContainerStyle(showOutput), isOfflineDocsBuild ? OFFLINE_OUTPUT_CLASSNAME : '')}>
            <Output nodeData={children[1]} />
          </div>
        </>
      )}
      {onlyInputSpecified && <Input nodeData={children[0]} />}
    </div>
  );
};

const IOToggle = styled.div`
  ${borderCodeStyle}
  border-color: var(--border-color);
  border-top: none;
`;

CodeIO.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};

export default CodeIO;
