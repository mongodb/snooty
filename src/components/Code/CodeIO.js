import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { cx, css as LeafyCSS } from '@leafygreen-ui/emotion';
import Icon from '@leafygreen-ui/icon';
import Button from '@leafygreen-ui/button';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { theme } from '../../theme/docsTheme';
import { baseCodeStyle, borderCodeStyle } from './styles/codeStyle';
import Input from '../Code/Input';
import Output from '../Code/Output';

const outputButtonStyling = LeafyCSS`
  padding: 0px;
  font-size: ${theme.fontSize.tiny};
  height: 24px;
  margin: 8px;
  font-weight: bold;
`;

const CodeIO = ({ nodeData: { children }, ...rest }) => {
  const needsIOToggle = children.length === 2;
  const onlyInputSpecified = children.length === 1;

  let initialOutputVisibility = true;
  if (needsIOToggle && children[1]?.options?.visible !== undefined) {
    initialOutputVisibility = !!children[1].options.visible;
  }
  const [showOutput, setShowOutput] = useState(initialOutputVisibility);
  const buttonText = showOutput ? 'HIDE OUTPUT' : 'VIEW OUTPUT';
  const arrow = showOutput ? 'ChevronUp' : 'ChevronDown';
  const outputBorderRadius = !showOutput ? theme.size.tiny : '0px';
  const singleInputBorderRadius = onlyInputSpecified ? theme.size.tiny : '0px';

  const handleClick = (e) => {
    if (showOutput) {
      setShowOutput(false);
    } else {
      setShowOutput(true);
    }
  };

  if (children.length === 0) {
    return null;
  }
  return (
    <div
      css={css`
        ${baseCodeStyle}

        // Inner div of LG component has a width set to 700px. Unset this as part of our
        // override for docs when the language switcher is being used.
        > div > div {
          border-bottom-right-radius: ${singleInputBorderRadius};
          border-bottom-left-radius: ${singleInputBorderRadius};
        }

        // Controls output code block and toggle view bar style
        > div {
          border-top-right-radius: 0px;
          border-top-left-radius: 0px;
          border-bottom-right-radius: ${outputBorderRadius};
          border-bottom-left-radius: ${outputBorderRadius};
          margin: 0px;
        }
      `}
    >
      {needsIOToggle && (
        <>
          <Input nodeData={children[0]} />
          <IOToggle>
            <Button
              role="button"
              className={cx(outputButtonStyling)}
              darkMode={false}
              disabled={false}
              onClick={handleClick}
              leftGlyph={<Icon glyph={arrow} fill="#FF0000" />}
            >
              {buttonText}
            </Button>
          </IOToggle>
          {showOutput && <Output nodeData={children[1]} />}
        </>
      )}
      {onlyInputSpecified && <Input nodeData={children[0]} />}
    </div>
  );
};

const IOToggle = styled.div`
  ${borderCodeStyle}
  border-top: none;
`;

CodeIO.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};

export default CodeIO;
