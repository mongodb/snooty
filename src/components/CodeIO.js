import React, { useState } from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from './ComponentFactory';
import { cx, css as LeafyCss } from '@leafygreen-ui/emotion';
import { css } from '@emotion/core';
import Icon from '@leafygreen-ui/icon';
import Button from '@leafygreen-ui/button';
import styled from '@emotion/styled';
import { theme } from '../theme/docsTheme';

const outputButtonStyling = LeafyCss`
  padding: 0px;
  font-size: 12px;
  height: 24px;
  margin: 8px;
  font-weight: bold;
`;

const CodeIO = ({ nodeData: { children }, ...rest }) => {
  const [showOutput, setShowOutput] = useState('none');
  const [buttonText, setButtonText] = useState('VIEW OUTPUT');
  const [arrow, setArrow] = useState('ChevronDown');
  let outputExists = showOutput === 'none' ? '4px' : '0px';

  const handleClick = (e) => {
    if (showOutput === 'inline') {
      setShowOutput('none');
      setButtonText('VIEW OUTPUT');
      setArrow('ChevronDown');
    } else {
      setShowOutput('inline');
      setButtonText('HIDE OUTPUT');
      setArrow('ChevronUp');
    }
  };

  return (
    <div
      css={css`
        display: table;
        margin: ${theme.size.default} 0;
        min-width: 150px;
        table-layout: fixed;
        width: 100%;

        // Inner div of LG component has a width set to 700px. Unset this as part of our
        // override for docs when the language switcher is being used.
        > div > div {
          width: unset;
          border-radius: 0px;
        }

        > div {
          border-top-right-radius: 0px;
          border-top-left-radius: 0px;
          border-bottom-right-radius: ${outputExists};
          border-bottom-left-radius: ${outputExists};
          margin: 0px;
        }

        // Override default LG Code language switcher font size
        button > div > div {
          font-size: ${theme.fontSize.default};
        }
      `}
    >
      <ComponentFactory {...rest} nodeData={children[0]} />
      <IOToggle>
        <Button
          className={cx(outputButtonStyling)}
          darkMode={false}
          disabled={false}
          onClick={handleClick}
          leftGlyph={<Icon glyph={arrow} fill="#FF0000" />}
        >
          {buttonText}
        </Button>
      </IOToggle>
      {showOutput === 'inline' && <ComponentFactory {...rest} nodeData={children[1]} />}
      {children.length !== 2 && children.map((child, i) => <ComponentFactory {...rest} key={i} nodeData={child} />)}
    </div>
  );
};

const IOToggle = styled.div`
  border: 1px solid #e7eeec;
  border-top: none;
`;

CodeIO.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};

export default CodeIO;
