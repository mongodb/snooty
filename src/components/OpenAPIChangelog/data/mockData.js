export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
