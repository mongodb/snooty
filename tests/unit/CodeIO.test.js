import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CodeIO from '../../src/components/Code/CodeIO';

// data for this component
import mockData from './data/CodeIO.test.json';

describe('CodeIO', () => {
  it('renders correctly', () => {
    const wrapper = render(<CodeIO nodeData={mockData.outputVisibleByDefault} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('closes and opens output code snippet on io button click when output is visible by default', () => {
    const wrapper = render(<CodeIO nodeData={mockData.outputVisibleByDefault} />);
    userEvent.click(wrapper.getByRole('button'));
    expect(wrapper.queryAllByText('hello world')).toHaveLength(0);
    userEvent.click(wrapper.getByRole('button'));
    expect(wrapper.getByText('hello world')).toBeTruthy();
  });

  it('opens and closes output code snippet on io button click when output is hidden by default', () => {
    const wrapper = render(<CodeIO nodeData={mockData.outputHiddenByDefault} />);
    userEvent.click(wrapper.getByRole('button'));
    expect(wrapper.getByText('hello world')).toBeTruthy();
    userEvent.click(wrapper.getByRole('button'));
    expect(wrapper.queryAllByText('hello world')).toHaveLength(0);
  });
});
