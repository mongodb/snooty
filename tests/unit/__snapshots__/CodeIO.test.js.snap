// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeIO renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: table;
  margin: 16px 0;
  min-width: 150px;
  table-layout: fixed;
  width: 100%;
}

.emotion-0 button>div>div {
  font-size: 14px;
}

.emotion-0>div>div {
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
}

.emotion-0>div {
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  margin: 0px;
}

.emotion-1 {
  display: table;
  margin: 16px 0;
  min-width: 150px;
  table-layout: fixed;
  width: 100%;
}

.emotion-1 button>div>div {
  font-size: 14px;
}

.emotion-1>div>div {
  display: grid;
  grid-template-columns: code panel;
}

.emotion-1>div {
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  display: grid;
}

.emotion-2 {
  border: 1px solid #E7EEEC;
  border-bottom: none;
  border-top-right-radius: 12px;
  border-top-left-radius: 12px;
}

.emotion-4 {
  padding: 10px;
  color: #5D6C74;
  font-size: 14px;
  margin-left: 5px;
  border-bottom: none;
}

.emotion-5 {
  border: 1px solid #E8EDEB;
  border-radius: 12px;
  overflow: hidden;
}

.emotion-6 {
  position: relative;
  display: grid;
  grid-template-areas: 'code panel';
  grid-template-columns: auto 38px;
  border-radius: inherit;
  z-index: 0;
}

.emotion-6:before,
.emotion-6:after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  height: 100%;
  width: 8px;
  border-radius: 100%;
  box-shadow: unset;
  -webkit-transition: box-shadow 100ms ease-in-out;
  transition: box-shadow 100ms ease-in-out;
}

.emotion-6:before {
  grid-column: 1;
  left: -8px;
}

.emotion-6:after {
  grid-column: 2;
}

.emotion-6:before,
.emotion-6:after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  height: 100%;
  width: 8px;
  border-radius: 100%;
  box-shadow: unset;
  -webkit-transition: box-shadow 100ms ease-in-out;
  transition: box-shadow 100ms ease-in-out;
}

.emotion-6:before {
  grid-column: 1;
  left: -8px;
}

.emotion-6:after {
  grid-column: 2;
}

.emotion-7 {
  grid-area: code;
  overflow-x: auto;
  border-radius: inherit;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  margin: 0;
  position: relative;
  -webkit-transition: box-shadow 100ms ease-in-out;
  transition: box-shadow 100ms ease-in-out;
  white-space: pre;
  border-color: #E8EDEB;
  background-color: #F9FBFA;
  color: #001E2B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 6px;
  padding-bottom: 6px;
}

@media only screen and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-7 {
    white-space: pre-wrap;
  }
}

@media only screen and (min-device-width: 813px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-7 {
    white-space: pre;
  }
}

.emotion-7:focus,
.emotion-7:active,
.emotion-7:focus-visible,
.emotion-7:focus-within {
  outline: none;
  box-shadow: 0 0 0 2px #0498EC inset;
}

.emotion-8 {
  color: inherit;
  font-size: 13px;
  font-family: 'Source Code Pro',Menlo,monospace;
  line-height: 24px;
  font-size: 13px;
}

.emotion-9 {
  border-spacing: 0;
  width: 100%;
}

.emotion-10 {
  background-color: #FEF7E3;
  background-image: none;
  background-attachment: fixed;
}

.emotion-10>td {
  border-top: 1px solid #FEF2C8;
}

.emotion-10+tr>td {
  border-top: 1px solid #FEF2C8;
}

.emotion-10+.emotion-10>td {
  border-top: 0;
}

.emotion-10:last-child>td {
  border-bottom: 1px solid #FEF2C8;
}

.emotion-11 {
  border-spacing: 0;
  vertical-align: top;
  padding: 0 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: right;
  padding-left: 8px;
  padding-right: 0;
  color: #944F01;
}

.emotion-12 {
  border-spacing: 0;
  vertical-align: top;
  padding: 0 16px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 4px;
  padding: 6px;
  border-left: solid 1px;
  background-color: #FFFFFF;
  border-color: #E8EDEB;
  z-index: 2;
  grid-area: panel;
}

.emotion-13 svg {
  width: 16px;
  height: 16px;
}

.emotion-14 {
  border: 1px solid #E7EEEC;
  border-top: none;
}

.emotion-16 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #F9FBFA;
  border-color: #889397;
  color: #1C2D38;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 36px;
  padding: 0px;
  font-size: 12px;
  height: 24px;
  margin: 8px;
  font-weight: bold;
}

.emotion-16:focus {
  outline: none;
}

.emotion-16[disabled],
.emotion-16:disabled {
  pointer-events: none;
}

.emotion-16:active,
.emotion-16:focus,
.emotion-16:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16:hover,
.emotion-16:active {
  color: #1C2D38;
  background-color: #FFFFFF;
  box-shadow: 0 0 0 3px #E8EDEB;
}

.emotion-16:focus {
  background-color: #FFFFFF;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-17 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-18 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  pointer-events: none;
  position: relative;
  z-index: 0;
  padding: 0 11px;
  gap: 6px;
}

.emotion-19 {
  color: #FF0000;
  color: #889397;
  height: 16px;
  width: 16px;
  justify-self: right;
}

.emotion-20>div>* {
  display: inline!important;
}

.emotion-20 * {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
  border-bottom-right-radius: 12px;
  border-bottom-left-radius: 12px;
}

.emotion-21 {
  border: 0;
  border-radius: 6px;
  overflow: hidden;
}

.emotion-23 {
  grid-area: code;
  overflow-x: auto;
  border-radius: inherit;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  margin: 0;
  position: relative;
  -webkit-transition: box-shadow 100ms ease-in-out;
  transition: box-shadow 100ms ease-in-out;
  white-space: pre;
  border: 0;
  background-color: #061621;
  color: #F9FBFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 6px;
  padding-bottom: 6px;
}

@media only screen and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-23 {
    white-space: pre-wrap;
  }
}

@media only screen and (min-device-width: 813px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-23 {
    white-space: pre;
  }
}

.emotion-23:focus,
.emotion-23:active,
.emotion-23:focus-visible,
.emotion-23:focus-within {
  outline: none;
  box-shadow: 0 0 0 2px #0498EC inset;
}

.emotion-26 {
  background-color: transparent;
  background-image: linear-gradient(90deg, #21313C, rgba(33,49,60,0));
  background-attachment: fixed;
}

.emotion-26>td {
  border-top: 1px solid #21313C;
}

.emotion-26+tr>td {
  border-top: 1px solid #21313C;
}

.emotion-26+.emotion-26>td {
  border-top: 0;
}

.emotion-26:last-child>td {
  border-bottom: 1px solid #21313C;
}

.emotion-27 {
  border-spacing: 0;
  vertical-align: top;
  padding: 0 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: right;
  padding-left: 8px;
  padding-right: 0;
  color: #E7EEEC;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div>
        <div
          class="emotion-2 emotion-3"
        >
          <div
            class="emotion-4"
          >
            Title of Example
          </div>
        </div>
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        >
          <pre
            class="emotion-7"
            tabindex="-1"
          >
            <code
              class="lg-highlight-hljs-light python emotion-8"
            >
              <table
                class="emotion-9"
              >
                <tbody>
                  <tr
                    class="emotion-10"
                  >
                    <td
                      class="emotion-11"
                    >
                      1
                    </td>
                    <td
                      class="emotion-12"
                    >
                      <span
                        class="lg-highlight-built_in"
                      >
                        print
                      </span>
                      (
                      <span
                        class="lg-highlight-string"
                      >
                        'hello world'
                      </span>
                      )
                    </td>
                  </tr>
                </tbody>
              </table>
            </code>
          </pre>
          <div
            class="emotion-13"
            data-testid="leafygreen-code-panel"
          />
        </div>
      </div>
    </div>
    <div
      class="emotion-14 emotion-15"
    >
      <button
        aria-disabled="false"
        class="emotion-16"
        data-leafygreen-ui="button"
        role="button"
        type="button"
      >
        <div
          class="emotion-17"
        />
        <div
          class="emotion-18"
        >
          <svg
            alt=""
            aria-hidden="true"
            class="emotion-19"
            fill="none"
            height="16"
            role="presentation"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M14.364 10.778a1 1 0 000-1.414l-4.95-4.95-.707-.707a1 1 0 00-1.414 0l-.707.707-4.95 4.95a1 1 0 000 1.414l.707.707a1 1 0 001.414 0L8 7.243l4.243 4.242a1 1 0 001.414 0l.707-.707z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          HIDE OUTPUT
        </div>
      </button>
    </div>
    <div
      class="emotion-20"
    >
      <div
        class="emotion-21"
      >
        <div
          class="emotion-6"
        >
          <pre
            class="emotion-23"
            tabindex="-1"
          >
            <code
              class="lg-highlight-hljs-dark python emotion-8"
            >
              <table
                class="emotion-9"
              >
                <tbody>
                  <tr
                    class="emotion-26"
                  >
                    <td
                      class="emotion-27"
                    >
                      1
                    </td>
                    <td
                      class="emotion-12"
                    >
                      hello world
                    </td>
                  </tr>
                </tbody>
              </table>
            </code>
          </pre>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
